be/src/runtime/stream_load/stream_load_executor.cpp

```diff
@@ -223,7 +223,16 @@ Status StreamLoadExecutor::commit_txn(StreamLoadContext* ctx) {
         request.__isset.txnCommitAttachment = true;
     }
 
-    return commit_txn_internal(request, rpc_timeout_ms, ctx);
+    auto st = commit_txn_internal(request, rpc_timeout_ms, ctx);
+    // service unavailable implies that the exception occurred in network transmission or Thrift RPC framework,
+    // rather than due to slow publish. Therefore, we can utilize the remaining timeout to retry.
+    if (st.is_service_unavailable()) {
```

Comment:
retry 3 times using a while loop?

Reply:
Not necessary, just increases the pressure on FE